src/main/java/techcourse/myblog/domain/User.java

```diff
@@ -0,0 +1,108 @@
+package techcourse.myblog.domain;
+
+import techcourse.myblog.domain.exception.UserArgumentException;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import java.util.regex.Pattern;
+
+import static techcourse.myblog.domain.exception.UserArgumentException.*;
+
+@Entity
+public class User {
+    private static final int MIN_NAME_LENGTH = 2;
+    private static final int MAX_NAME_LENGTH = 10;
+    private static final int MIN_PASSWORD_LENGTH = 8;
+    private static final Pattern KOREAN_ENGLISH_PATTERN = Pattern.compile("^[(ㄱ-ㅎ가-힣a-zA-Z)]+$");
+    private static final Pattern LOWER_CASE_PATTERN = Pattern.compile("[(a-z)]+");
+    private static final Pattern UPPER_CASE_PATTERN = Pattern.compile("[(A-Z)]+");
+    private static final Pattern NUMBER_PATTERN = Pattern.compile("[(0-9)]+");
+    private static final Pattern SPECIAL_CHARACTER_PATTERN =
+            Pattern.compile("[ `~!@#[$]%\\^&[*]\\(\\)_-[+]=\\{\\}\\[\\][|]'\":;,.<>/?]+");
+    private static final Pattern KOREAN_PATTERN = Pattern.compile("[(ㄱ-ㅎ가-힣)]+");
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String name;
+    private String email;
+    private String password;
+
+    public User() {
```

Comment:
JPA가 필요로 하는 기본 생성자는 private으로 만들어야 다른 개발자의 오용을 방지할 수 있어요. 아래의 글은 가볍게 참고만 해주세요.
https://goodyhlee.wordpress.com/2016/06/20/177

```suggestion
    private User() {
```

Reply:
private로 변경하였습니다!

한가지 궁금한 점이 있는데 링크에 있는 @PersistenceConstructor의 역할이 궁금합니다. @PersistenceConstructor가 리플렉션시 사용되는 생성자를 지정해주는 것이라고 생각했는데 기존 코드에서 default 생성자를 지우고 다른 생성자에 @PersistenceConstructor를 붙여도 정상 동작하지 않는 것 같습니다. 자료를 찾아봐도 잘 이해가 안되서 질문드립니다.

Reply:
아래의 글을 참고하세요.
https://stackoverflow.com/questions/52059543/does-spring-data-jpas-persistenceconstructor-annotation-work-in-combination-wi

Reply:
hibernate와는 관련없는 어노테이션이였군요..